kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/OpenIDConnectionUtils.java

```diff
@@ -242,6 +240,12 @@ static boolean persistKubeConfigWithUpdatedToken(Config currentConfig, Map<Strin
    */
   public static boolean persistKubeConfigWithUpdatedAuthInfo(Config currentConfig, Consumer<AuthInfo> updateAction)
       throws IOException {
+    AuthInfo authInfo = new AuthInfo();
+    authInfo.setAuthProvider(new AuthProviderConfig(new HashMap<>(2), currentConfig.getAuthProvider().getName()));
```

Comment:
This leads to a NPE if the current config has no auth provider set

Reply:
Currently working on some refactors and fixes

Reply:
> This leads to a NPE if the current config has no auth provider set

I think the OpenIDConnectionUtils wouldn't be invoked if no auth provider set, right?

Reply:
Probably not, but that's unclear in tests.
Also the method is reused elsewhere (i.e. this one is called from the OpenShiftOAuthInterceptor) because it was public.

Anyway, this might have some ramifications. I'll work on refactoring some of this stuff and maybe moving it elsewhere to avoid latent bugs.